File: ModelInterfacesClassLibrary/Phonebook/PhbkPhone/PhbkPhoneViewPage.interface.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using Newtonsoft.Json;
using Newtonsoft.Json.Serialization;
using ModelInterfacesClassLibrary.Phonebook.PhbkPhone;

namespace ModelInterfacesClassLibrary.Phonebook.PhbkPhone {
    public interface IPhbkPhoneViewPage
    {
        [JsonProperty(PropertyName = "page")]
        int page { get; set; }

        [JsonProperty(PropertyName = "pagesize")]
        int pagesize { get; set; }

        [JsonProperty(PropertyName = "pagecount")]
        int pagecount { get; set; }

        [JsonProperty(PropertyName = "total")]
        int total { get; set; }

        [JsonProperty(PropertyName = "items")]
        IList<IPhbkPhoneView> items { get; set; }
    }
}

File: AngularMatPhonebookSolution/projects/phonebook-interfaces/src/lib/phbk-phone/interfaces/phbk-phone-view-page.interface.ts
import { IPhbkPhoneView } from './phbk-phone-view.interface';

export interface IPhbkPhoneViewPage {
        page : number; // int
        pagesize : number; // int
        pagecount : number; // int
        total : number; // int
        items ?: IPhbkPhoneView[];
}


